addons/auth_signup/models/res_users.py

```diff
@@ -337,6 +339,18 @@ def _prepare_new_device_notice_values(self):
                 values['useros'] = request.httprequest.user_agent.platform.capitalize()
         return values
 
+    def _alert_untrusted_location(self):
+        if not str2bool(self.env['ir.config_parameter'].sudo().get_param("auth_signup.alert_untrusted_location", 'true')):
+            return
+        # Fallback on ``_alert_new_device`` with a no readonly cursor
+        if self.env.cr.readonly:
+            self.env.cr.rollback()
+            cursor = self.env.registry.cursor(readonly=False)
+        else:
+            cursor = nullcontext(self.env.cr)
+        with cursor as cr:
+            self.with_env(self.env(cr=cr))._alert_new_device()
```

Comment:
It's a bit a pity to open again a read-write cursor while we already open one in `_update_device`.
Isn't there a way to re-work the code so it would re-use the same read-write cursor that was opened in `_update_device` ?